docs/core/deploying/trimming/snippets/MyLibrary/Class1.cs

```diff
@@ -0,0 +1,222 @@
+    static Type type;
+    static void UseMethodsHelper()
+    {
+        // warning IL2077: MyLibrary.UseMethodsHelper(Type): 'type' argument does not satisfy
+        // 'DynamicallyAccessedMemberTypes.PublicMethods' in call to
+        // 'MyLibrary.UseMethods(Type)'.
+        // The field 'System.Type MyLibrary::type' does not have matching annotations.
+        UseMethods(type);
+    }
+    // </snippet_UMH>
+}
+
+
+public class MyLibrary7
+{
+    // <snippet_AD1>
+    class TypeCollection
+    {
+        Type[] types;
+
+        // Ensure that only types with preserved constructors are stored in the array
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]
+        public Type this[int i]
+        {
+            // warning IL2063: TypeCollection.Item.get: Value returned from method
+            // 'TypeCollection.Item.get' can't be statically determined and may not meet
+            // 'DynamicallyAccessedMembersAttribute' requirements.
+            get => types[i];
+            set => types[i] = value;
+        }
+    }
+
+    class TypeCreator
+    {
+        TypeCollection types;
+
+        public void CreateType(int i)
+        {
+            types[i] = typeof(TypeWithConstructor);
+            Activator.CreateInstance(types[i]); // No warning!
+        }
+    }
+
+    class TypeWithConstructor
+    {
+    }
+    // </snippet_AD1>
+}
+
+public class MyLibrary8
+{
+    // <snippet_AD2>
+    class TypeCollection
+    {
+        Type[] types;
+
+        // Ensure that only types with preserved constructors are stored in the array
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]
+        public Type this[int i]
+        {
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2063",
+                Justification = "The list only contains types stored through the annotated setter.")]
+            get => types[i];
+            set => types[i] = value;
+        }
+    }
+
+    class TypeCreator
+    {
+        TypeCollection types;
+
+        public void CreateType(int i)
+        {
+            types[i] = typeof(TypeWithConstructor);
+            Activator.CreateInstance(types[i]); // No warning!
+        }
+    }
+
+    class TypeWithConstructor
+    {
+    }
+    // </snippet_AD2>
+}
+
+public class MyLibrary11
+{
+    // <snippet_AD3>
+    [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2063",
+        // Invalid justification and suppression: property being non-reflectively
+        // used by the app doesn't guarantee that the property will be available
+        // for reflection. Properties that are not visible targets of reflection
+        // are already optimized away with Native AOT trimming and may be
+        // optimized away for non-native deployment in the future as well.
+        Justification = "*INVALID* Only need to serialize properties that are used by"
+                        + "the app. *INVALID*")]
```

Comment:
The formatting of this code snippet in the rendered view is broken - we will probably need to move the comment above the attribute to fix it.